### PR TITLE
refactor: use PortexRecordBase column operations to replace to_builtin

### PR DESCRIPTION
**Pull request by Lee-000**

### CODE REVIEW

**Comment by GravitiRob** [comment]
Task linked: [CU-2ht81m0 使用 RecordBase 提供的列操作接口代替 to_builtin 接口](https://app.clickup.com/t/2ht81m0)

File: graviti/dataframe/frame.py [hunk 1]
```diff
@@ -718,8 +716,7 @@ def _process_record(
     ) -> Tuple[Callable[[Any], Any], bool]:
         processors: Dict[str, Callable[[Any], Any]] = {}
         need_process, sub_need_process = False, False
-        fields = schema.to_builtin().fields
-        for name, field in fields.items():
+        for name, field in schema.fields.items():
```
Comment (linjiX) [inline graviti/dataframe/frame.py:719]: why we need `.fields` here?

File: graviti/dataframe/sql/array.py [hunk 2]
```diff
@@ -177,7 +177,7 @@ class DataFrame(SearchContainer):
     schema: pt.PortexRecordBase
 
     def __getitem__(self, key: str) -> SearchContainer:
-        field: pt.PortexType = self.schema.to_builtin().fields[key]  # type: ignore[attr-defined]
+        field: pt.PortexType = self.schema[key]
```
Comment (linjiX) [inline graviti/dataframe/sql/array.py:180]: do we need the type hint here?

**Review by linjiX — CHANGES_REQUESTED** [review]

File: graviti/dataframe/frame.py [hunk 3]
```diff
@@ -718,8 +716,7 @@ def _process_record(
     ) -> Tuple[Callable[[Any], Any], bool]:
         processors: Dict[str, Callable[[Any], Any]] = {}
         need_process, sub_need_process = False, False
-        fields = schema.to_builtin().fields
-        for name, field in fields.items():
+        for name, field in schema.items():
```
Comment (linjiX) [inline graviti/dataframe/frame.py:719]: do we need to change the type hint of schema in the signature of `_process_record`?

**Review by linjiX — APPROVED** [review]

**Pull request merged**